### PR TITLE
MGDOBR-992: BridgeIngressServiceTest/BridgeExecutorServiceTest: Fix disabled tests

### PR DESCRIPTION
**Pull request by manstis**
[https://issues.redhat.com/browse/MGDOBR-992](https://issues.redhat.com/browse/MGDOBR-992)

## PR owner checklist

Please make sure that your PR meets the following requirements:

- [x] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config).  
  *Run `mvn clean verify -DskipTests` so that imports are correctly set.*
- [x] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [x] The layers in the `kustomize` folder have been updated accordingly.
- [x] All new functionality is tested
- [x] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

## PR reviewer(s) checklist

- [ ] Check if code and Github action workflows have been modified and if the modification is safe
- [ ] If the modification is safe, add the `safe to test` label

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

- <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

- <b>Rebase the pull request</b>  
  Comment with: `/rebase`.
- <b>Deploy BOT</b>
  Comment with `/deploy <dev,stable> when the PR has been merged to deploy to a target environment.

</details>

### CODE REVIEW

**Review by r00ta — APPROVED** [review]
Would you wait for https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/1006 ?  I don't have much time left before my PTO and I'm still struggling to understand why the e2e tests are not working - a rebase would be a bit time consuming/wait for CI blablabla

**Review by sutaakar — APPROVED** [review]

**Review by robpblake — APPROVED** [review]
Thank you @manstis . Nice job finding an approach that doesn't require quite so much mocking.

Would you mind creating us a task in the backlog to review the operator tests to see if we're following best practice and that our unit tests should definitely be unit tests. This test suite has grown organically over time, so might be a good chance to take stock given our increasing team experience of working with the operator.

cheers,

Rob

**Comment by manstis** [comment]
@robpblake 

> Would you mind creating us a task in the backlog to review the operator tests to see if we're following best practice and that our unit tests should definitely be unit tests. This test suite has grown organically over time, so might be a good chance to take stock given our increasing team experience of working with the operator.

https://issues.redhat.com/browse/MGDOBR-1002

File: shard-operator/src/test/java/com/redhat/service/smartevents/shard/operator/BridgeExecutorServiceTest.java [hunk 1]
```diff
@@ -169,43 +181,77 @@ public void testBridgeExecutorDeletion() {
...
+                });
+
+        Awaitility.await()
+                .atMost(Duration.ofMinutes(2))
+                .pollInterval(Duration.ofSeconds(5))
+                .untilAsserted(
+                        () -> {
+                            BridgeExecutor bridgeExecutor = fetchBridgeExecutor(dto);
+                            assertThat(bridgeExecutor.getStatus().isReady()).isTrue();
+                        });
```
Comment (r00ta) [inline shard-operator/src/test/java/com/redhat/service/smartevents/shard/operator/BridgeExecutorServiceTest.java:228]: Nitpick: create a method to remove all the duplicated 
```
Awaitility.await()
                .atMost(Duration.ofMinutes(2))
                .pollInterval(Duration.ofSeconds(5))
                .untilAsserted(
```
?

File: shard-operator/src/test/java/com/redhat/service/smartevents/shard/operator/BridgeIngressServiceTest.java [hunk 2]
```diff
@@ -153,30 +158,60 @@ public void testBridgeIngressDeletion() {
...
+
+        Awaitility.await()
+                .atMost(Duration.ofMinutes(2))
+                .pollInterval(Duration.ofSeconds(5))
+                .untilAsserted(
+                        () -> {
+                            BridgeIngress fetched = fetchBridgeIngress(dto);
+                            assertThat(fetched.getStatus().isReady()).isTrue();
+                        });
+
```
Comment (r00ta) [inline shard-operator/src/test/java/com/redhat/service/smartevents/shard/operator/BridgeIngressServiceTest.java:194]: same here

**Review by r00ta — APPROVED** [review]

**Pull request merged**